packages/eslint-plugin/src/util/collectUnusedVariables.ts

```diff
@@ -739,6 +744,7 @@ function isUsedVariable(variable: TSESLint.Scope.Variable): boolean {
     return (
       ref.isRead() &&
       !forItself &&
+      !(!isImportedAsType && referenceContainsTypeQuery(ref.identifier)) &&
```

Comment:
With the current logic, this is allowed:

```TS
import type { valueVariable } from './a';

export type Foo = typeof valueVariable;
```

Not sure whether it's the right behavior, but there is a test for this already:
https://github.com/typescript-eslint/typescript-eslint/blob/33f723f841badf6c34c5a7092eb2cd2438de9f77/packages/eslint-plugin/tests/rules/no-unused-vars/no-unused-vars.test.ts#L830-L834

Reply:
Yeah, this is fine. The variable is imported as a type and used as a type. There's no unused runtime stuff.